plugins/modules/ndo_l3out_node_group_policy.py

```diff
@@ -0,0 +1,316 @@
+            )
+
+        if mso.existing:
+            proposed_payload = copy.deepcopy(mso.existing)
+
+            if description is not None and mso.existing.get("description") != description:
+                ops.append(dict(op="replace", path=node_group_policy_path + "/description", value=description))
+                proposed_payload["description"] = description
+
+            if (
+                node_routing_policy is not None
+                and l3out_node_routing_policy_object
+                and mso.existing.get("nodeRoutingPolicyRef") != l3out_node_routing_policy_object.details.get("uuid")
+            ):
+                ops.append(
+                    dict(op="replace", path=node_group_policy_path + "/nodeRoutingPolicyRef", value=l3out_node_routing_policy_object.details.get("uuid"))
+                )
+                proposed_payload["nodeRoutingPolicyRef"] = l3out_node_routing_policy_object.details.get("uuid")
+            elif node_routing_policy == "" and mso.existing.get("nodeRoutingPolicyRef"):
+                ops.append(dict(op="remove", path=node_group_policy_path + "/nodeRoutingPolicyRef"))
+                proposed_payload.pop("nodeRoutingPolicyRef", None)
+
+            if bfd_multi_hop_authentication == "enabled":
+                if not mso.existing.get("bfdMultiHop"):
+                    proposed_payload["bfdMultiHop"] = dict()
+                    ops.append(dict(op="replace", path=node_group_policy_path + "/bfdMultiHop", value=dict()))
+
+                if mso.existing.get("bfdMultiHop", {}).get("keyID") != bfd_multi_hop_key_id:
+                    ops.append(dict(op="replace", path=node_group_policy_path + "/bfdMultiHop/keyID", value=bfd_multi_hop_key_id))
+                    proposed_payload["bfdMultiHop"]["keyID"] = bfd_multi_hop_key_id
+
+                ops.append(dict(op="replace", path=node_group_policy_path + "/bfdMultiHop/value", value=bfd_multi_hop_key))
+                proposed_payload["bfdMultiHop"]["value"] = bfd_multi_hop_key
+
+            elif bfd_multi_hop_authentication == "disabled" and mso.existing.get("bfdMultiHop"):
+                proposed_payload.pop("bfdMultiHop", None)
+                ops.append(dict(op="remove", path=node_group_policy_path + "/bfdMultiHop"))
+
+            if target_dscp is not None and mso.existing.get("targetDscp") != target_dscp:
+                ops.append(dict(op="replace", path=node_group_policy_path + "/targetDscp", value=target_dscp))
+                proposed_payload["targetDscp"] = target_dscp
+
+            mso.sanitize(proposed_payload, collate=True)
+
+        else:
+            payload = dict(name=name)
+
+            if description:
+                payload["description"] = description
+
+            if node_routing_policy and l3out_node_routing_policy_object:
+                payload["nodeRoutingPolicyRef"] = l3out_node_routing_policy_object.details.get("uuid")
+
+            if bfd_multi_hop_authentication == "enabled":
```

Comment:
How do we deal with BFD multihop enabled without authentication situation? Currently I do not see this possible with your current config

Which would be represented in config as:

```
            "bfdMultiHop": {
              "authEnabled": false
            },